tests/test_configurable_lookback_window.py

```diff
@@ -0,0 +1,86 @@
+"""Test tap configurable properties. Specifically the lookback_window"""
+import re
+import os
+from datetime import datetime as dt
+from datetime import timedelta
+
+from tap_tester import menagerie, connections, runner
+
+from base import BaseTapTest
+
+
+class ConversionWindowBaseTest(BaseTapTest):
+    """
+    Test tap's sync mode can execute with valid lookback_window values set.
+    Validate setting the lookback_window configurable property.
+    Test Cases:
+    Verify connection can be created, and tap can discover and sync with a lookback window
+    when passed in config else takes default value.
+    """
+    lookback_window = ''
+
+    def name(self):
+        return f"tt_stripe_lookback_window_{self.lookback_window}"
+
+    def get_properties(self):
+        """Configurable properties, with a switch to override the 'start_date' property"""
+        return_value = {
+            'start_date':dt.strftime(dt.utcnow() - timedelta(seconds=600), self.START_DATE_FORMAT),
+            'lookback_window': self.lookback_window,
+            'account_id': os.getenv('TAP_STRIPE_ACCOUNT_ID'),
+            'client_secret': os.getenv('TAP_STRIPE_CLIENT_SECRET')
+        }
+        return return_value
+
+    def run_test(self):
+        """
+        Testing that basic sync functions without Critical Errors when
+        a valid lookback_window is set.
+        """
+        print("Configurable Properties Test (lookback_window)")
+
+        conn_id = connections.ensure_connection(self)
+
+        streams_to_test = {'balance_transactions'}
+
+        # Run a discovery job
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # Perform table and field selection...
+        core_catalogs = [catalog for catalog in found_catalogs
+                         if catalog['stream_name'] in streams_to_test]
+        report_catalogs = [catalog for catalog in found_catalogs
+                           if catalog['stream_name'] in streams_to_test]
+        # select all fields for core streams and...
+        self.select_all_streams_and_fields(conn_id, core_catalogs, select_all_fields=True)
+
+        # set state to ensure conversion window is used
+        today_datetime = int((dt.utcnow() - timedelta(days=1)).timestamp())
+        today_datetime = int(dt.utcnow().timestamp())
+        
+        initial_state = {
+            'currently_syncing': None,
+            'bookmarks': {stream: {'created': today_datetime}
+                          for stream in streams_to_test
+                          if stream == 'balance_transactions'}
+        }
+        menagerie.set_state(conn_id, initial_state)
```

Comment:
If this is only setting state for 1 stream, it doesn't need to be a dictionary comprehension like this.